document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -123,13 +124,36 @@ public boolean upsert(Key key, Document document) throws IOException {
         LOGGER.debug("Write result: " + writeResult.toString());
       }
 
-      return writeResult.getModifiedCount() > 0;
+      return (writeResult.getUpsertedId() != null || writeResult.getModifiedCount() > 0);
```

Comment:
for insert modifiedcount will be 0 so the upsertId condition is added I assume ?

Reply:
Yes. But, this is not needed now.